apps/studio/data/edge-functions/edge-functions-deploy-mutation.ts

```diff
@@ -0,0 +1,87 @@
+import { useMutation, UseMutationOptions, useQueryClient } from '@tanstack/react-query'
+import { toast } from 'sonner'
+
+import { handleError, constructHeaders } from 'data/fetchers'
+import { API_URL } from 'lib/constants'
+import type { ResponseError } from 'types'
+import { edgeFunctionsKeys } from './keys'
+
+export type EdgeFunctionsDeployVariables = {
+  projectRef: string
+  metadata: {
+    entrypoint_path?: string
+    import_map_path?: string
+    name?: string
+    static_patterns?: string[]
+    verify_jwt?: boolean
+  }
+  files: { name: string; content: string }[]
+}
+
+export async function deployEdgeFunction({
+  projectRef,
+  metadata,
+  files,
+}: EdgeFunctionsDeployVariables) {
+  if (!projectRef) throw new Error('projectRef is required')
+
+  const formData = new FormData()
+  formData.append('metadata', JSON.stringify(metadata))
+
+  files.forEach((file) => {
+    const blob = new Blob([file.content], { type: 'text/plain' })
+    formData.append('file', blob, file.name)
+  })
+
+  const response = await fetch(
+    `${API_URL?.replace('/platform', '')}/v1/projects/${projectRef}/functions/deploy?slug=${metadata.name}`,
```

Comment:
do we need this part tho? `API_URL?.replace('/platform', '')`

we can't just do `/v1`? Similar to https://github.com/supabase/supabase/blob/master/apps/studio/data/edge-functions/edge-functions-query.ts#L18